packages/start/vite/plugin.js

```diff
@@ -671,44 +672,48 @@ export default function solidStart(options) {
   ].filter(Boolean);
 }
 
+/**
+ * @returns {import('vite').Plugin}
+ */
 function islands() {
   return {
     name: "solid-start-islands",
-    load(id) {
-      if (id.endsWith("?island")) {
-        return {
-          code: `
-            import Component from '${id.replace("?island", "")}';
+    async transform(code, id, opts) {
+      if (code.includes("unstable_island")) {
+        const ssr = opts?.ssr === true;
 
-            window._$HY.island("${id.slice(process.cwd().length)}", Component);
+        /** @type {Record<string, string>} */
+        const islandComponentIds = Object.fromEntries(
+          await Promise.all(
+            [
+              ...code.matchAll(
+                /const ([A-Za-z_]+) = unstable_island\(\(\) => import\((("([^"]+)")|('([^']+)'))\)\)/g
+              )
+            ].map(async ([a, b, c]) => {
+              c = c.slice(1, -1);
+              const resolved = await this.resolve(c, id);
+              if (!resolved) {
+                throw new Error(`Cannot resolve ${c} from ${id}`);
+              }
+              const componentId = createHash(resolved.id);
```

Comment:
I used a md5 hash. But any function is acceptable as long as it is a injective function.
In dev, maybe `resolve.id` can be simply used.
In prod, I guess the original path needs to be hidden.